tests/integration/gh-6970/src/test/java/io/helidon/tests/integration/tracerbaggage/MainTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tests.integration.tracerbaggage;
+
+import io.helidon.tracing.HeaderConsumer;
+import io.helidon.tracing.HeaderProvider;
+import io.helidon.tracing.Span;
+import io.helidon.tracing.Tracer;
+import io.helidon.webserver.WebServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+class MainTest {
+
+    private static WebServer webServer;
+
+    @BeforeAll
+    static void startTheServer() {
+        webServer = Main.startServer().await(10, TimeUnit.SECONDS);
+
+    }
+
+    @AfterAll
+    static void stopServer() {
+        if (webServer != null) {
+            webServer.shutdown()
+                    .await(10, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Test for: https://github.com/helidon-io/helidon/issues/6970
+     */
+    @Test
+    void baggageCanaryMinimal() {
+        Tracer tracer = Tracer.global();
+        Span span = tracer.spanBuilder("baggageCanaryMinimal").start();
+        // Set baggage and confirm that it's known in the span
+        span.baggage("fubar", "1");
+        assertThat(span.baggage("fubar").orElse(null), is("1"));
+
+        // Inject the span (context) into the consumer
+        HeaderConsumer consumer = HeaderConsumer
+                .create(new TreeMap<>(String.CASE_INSENSITIVE_ORDER));
+        tracer.inject(span.context(), HeaderProvider.empty(), consumer);
+
+        // Check baggage propagated
+        List<String> result = new ArrayList<>();
+        consumer.keys().forEach(result::add);
+        assertThat(result, hasItem(containsString("fubar")));
```

Comment:
might be worth testing the value also

Reply:
A stronger test would be to use `Runtime` to spawn another process with a `ClientMain` that calls into the server.

Reply:
Added value test